tests/functional/adapter/test_unique_tmp_table_suffix.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+from dbt.contracts.results import RunStatus
+from dbt.tests.util import run_dbt
+
+models__unique_tmp_table_suffix_sql = """
+{{ config(
+        materialized='incremental',
+        incremental_strategy='insert_overwrite',
+        partitioned_by=['date_column'],
+        unique_tmp_table_suffix=True
+    )
+}}
+select
+    random() as rnd,
+    cast(from_iso8601_date('{{ var('logical_date') }}') as date) as date_column
+"""
+
+
+class TestUniqueTmpTableSuffix:
```

Comment:
The issue I have with this test is that it does not test that indeed it created an unique suffix. But I am not sure we have a way to check this 🙁

Reply:
It will be ideal to check this, how do we do it? it's pretty tricky right? because tmp tables are drop at the end of the run :( Could we catch maybe some logs to maybe verify this? I've no better idea.

Reply:
Maybe it is possible by using the caplog fixture to get the logs https://github.com/dbt-athena/dbt-athena/blob/e458b59d3b0c160c0aef48b06a5ff2596f9cd00b/tests/conftest.py#L54  and then indeed use your solution to check the logs. This is tricky and not that clean but it should not be too hard to implement.

Reply:
Yes, it would make sense to test if the second create statement with a regex to match `unique_tmp_table_suffix__dbt_tmp_<UUID>`...
I'm trying to get the log from `[0m16:40:06.409303 [debug] [Thread-1 (]: dbt.adapters.athena.constants adapter: Running Athena query:
` logger using the caplog fixture of pytest but honestly struggling a bit.

Within `test__unique_tmp_table_suffix`, `caplog.text` seems to always be empty. I also tried using `dbt_debug_caplog`, but same result; I'm not able to get the logged outputted by the `AdapterLogger` during the run. Any idea what I'm missing? or any guidance you could provide me?
Thanks!

Reply:
I tried locally with this code sample and it works to get the logs 🔥 

```python
with tempfile.NamedTemporaryFile() as f_in:
    model_run = run_dbt([
        "run",
        "--select", relation_name,
        "--log-level", "debug", "--log-path", f_in.name, "--log-format", "json"
    ])
            
    logs = f_in.readlines()

    # Check the list of logs for what you want about the temp table
```

Reply:
@pierrebzl could you add that in your integration tests and parse the "tmp" table, and check if the tmp table is unique?

Reply:
Hum in fact sorry it does not work sorry I double checked I see the logs but only on the pytest console 😢

Reply:
Very nice, yes your previous message put me on the right path. 
I'll try to wrap something around this and come back to you. thanks for the help!

Reply:
I've updated the test suing your input, not the cleanest solution but it seems to fulfill its purpose.
Let me know what you think.